src/test/java/org/ays/step_definitions/Login.java

```diff
@@ -7,34 +7,183 @@
 import org.ays.browser.AysPageActions;
 import org.ays.pages.LoginPOM;
 import org.ays.configuration.AysConfigurationProperty;
-import org.testng.Assert;
+import org.ays.utilities.AysLocalizationUtil;
+import org.ays.utilities.AysLocaleStorageUtil;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
 
 public class Login {
 
-    private final LoginPOM loginPOM = new LoginPOM();
-    private final AysPageActions pageActions = new AysPageActions();
+    private final LoginPOM loginPOM;
+    private final AysPageActions pageActions;
+    private final AysLocaleStorageUtil localeStorageUtil;
+
+
+    public Login() {
+
+        this.loginPOM = new LoginPOM();
+        this.pageActions = new AysPageActions();
+        this.localeStorageUtil = new AysLocaleStorageUtil();
+
+    }
 
     @Given("Open the institution login page")
-    public void open_the_institution_login_page() {
-        this.pageActions.getWebDriver().get(AysConfigurationProperty.Ui.URL);
-        this.pageActions.getWebDriver().manage().window().maximize();
+    public void openTheInstitutionLoginPage() {
+        pageActions.getWebDriver().get(AysConfigurationProperty.Ui.URL);
+        pageActions.getWebDriver().manage().window().maximize();
+    }
+
+    @When("Enter the email address and password")
+    public void enterTheEmailAddressAndPassword() {
+        pageActions.sendKeysMethod(loginPOM.getLoginEmailAddress(), AysConfigurationProperty
+                .TestVolunteerFoundation.Admin.EMAIL_ADDRESS);
+        pageActions.sendKeysMethod(loginPOM.getLoginPassword(), AysConfigurationProperty
+                .TestVolunteerFoundation.Admin.PASSWORD);
+    }
+
+    @And("Click on the Login button")
+    public void clickOnTheLoginButton() {
+        pageActions.clickMethod(loginPOM.getLoginButton());
+    }
+
+    @Then("The username should be displayed on the homepage after successful login")
+    public void theUsernameShouldBeDisplayedOnTheHomepageAfterSuccessfulLogin() {
+        pageActions.waitUntilVisible(loginPOM.getUsername());
+        assertTrue(pageActions.isPresent(loginPOM.getUsername()));
+    }
+
+    @And("accessToken and refreshToken should be stored in localStorage")
+    public void accessTokenAndRefreshTokenShouldBeStoredInLocalStorage() {
+        pageActions.waitFor(2);
```

Comment:
selam, buralarda explicit wait işe yariyorsa onu kullansak daha iyi olabilir, belki denemissindir ama gözünden kaçtıysa diye belirtmek istedim.

Reply:
selam Hatice, burda element olmadığı için mecbur thread sleep kullandım. Localestorage ın yüklenmesi için biraz zamana ihtiyaç oluyor o yüzden explicit implicit kullanamadım